ahcore/models/jit_model.py

```diff
@@ -1,24 +1,231 @@
 from pathlib import Path
-from typing import Any
+from typing import Any, Callable
 
 import torch
 from torch.jit import ScriptModule, load
 from torch.nn import Module
 
+from ahcore.utils.types import AllModelTypes, AllOutputModes, FMEmbedType, FMOutputMode, JitModelType, JitOutputMode
+
 
 class AhcoreJitModel(ScriptModule):
-    def __init__(self, model: ScriptModule) -> None:
+    """
+    Base class for the jit compiled models in Ahcore.
+    """
+    def __init__(self, model: ScriptModule, model_type: AllModelTypes, output_mode: AllOutputModes) -> None:
+        """
+        Constructor for the AhcoreJitModel class.
+
+        Parameters
+        ----------
+        model: ScriptModule
+            The jit compiled model.
+
+        model_type: AllModelTypes
+            The type of the model.
+
+        output_mode: AllOutputModes
+            The output mode of the model. This is used to determine the forward function of the model.
+
+        Returns
+        -------
+        None
+        """
         super().__init__()  # type: ignore
         self._model = model
+        self._model_type = model_type
+        self._output_mode = output_mode
+        self._forward_function: Callable[[torch.Tensor], torch.Tensor]
 
     @classmethod
-    def from_jit_path(cls, jit_path: Path) -> Any:
+    def from_jit_path(cls, jit_path: Path, model_type: AllModelTypes, output_mode: AllOutputModes) -> Any:
+        """
+        Load a jit compiled model from a file path.
+
+        Parameters
+        ----------
+        jit_path : Path
+            The path to the jit compiled model.
+
+        model_type: AllModelTypes
+            The type of the model.
+
+        output_mode: AllOutputModes
+            The output mode of the model. This is used to determine the forward function of the model.
+
+        Returns
+        -------
+        An instance of the AhcoreJitModel class.
+        """
         model = load(jit_path)  # type: ignore
-        return cls(model)
+        return cls(model, model_type, output_mode)
 
     def extend_model(self, modules: dict[str, Module]) -> None:
+        """
+        Add modules to a jit compiled model.
+
+        Parameters
+        ----------
+        modules : dict[str, Module]
+            A dictionary of modules to add to the model.
+
+        Returns
+        -------
+        None
+        """
         for key, value in modules.items():
             self._model.add_module(name=key, module=value)
 
     def forward(self, x: torch.Tensor) -> Any:
+        return self._forward_function(x)
```

Comment:
How can this function be set?

Reply:
Right, I will implement an abstract function here which will be overwritten by each child.